src/java/org/apache/cassandra/service/accord/async/AsyncLoader.java

```diff
@@ -83,7 +83,11 @@ private static <K, V, S extends AccordSafeState<K, V>> void referenceAndAssemble
                                                                                                 List<AsyncChain<?>> listenChains)
     {
         S safeRef = cache.acquire(key);
-        context.put(key, safeRef);
+        if (context.putIfAbsent(key, safeRef) != null)
```

Comment:
this is for double counting the ref?

Reply:
right, this is to prevent us corrupting the count if we have somehow asked for the same key twice. I should probably add a NoSpamLogger at warn as well, as this is obviously a (minor, now) bug.